src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/linalg/FourierLinearOperatorNDArrayUnitTest.java

```diff
@@ -0,0 +1,128 @@
+    public Object[][] getTestDataWithoutZeroPadding() {
+        return new Object[][] {
+            {4,
+                new double[] {1.0, 1.0, 1.0},
+                new double[] {1.0, 2.0, 3.0, 4.0},
+                new double[] {1.0, 2.0, 3.0, 4.0}},
+            {4,
+                new double[] {0.0, 1.0, 1.0},
+                new double[] {1.0, 2.0, 3.0, 4.0},
+                new double[] {-1.5, -0.5, 0.5, 1.5}},
+            {4,
+                new double[] {-1.0, 3.0, 4.0},
+                new double[] {1.0, 2.0, 3.0, 4.0},
+                new double[] {-7.5, -3.5, -1.5, 2.5}},
+            {5,
+                new double[] {1.0, 1.0, 1.0},
+                new double[] {1.0, 2.0, 3.0, 4.0, 5.0},
+                new double[] {1.0, 2.0, 3.0, 4.0, 5.0}},
+            {5,
+                new double[] {0.0, 1.0, 1.0},
+                new double[] {1.0, 2.0, 3.0, 4.0, 5.0},
+                new double[] {-2.0, -1.0, 0.0, 1.0, 2.0}},
+            {5,
+                new double[] {1.0, 2.0, 1.0},
+                new double[] {1.0, 2.0, 3.0, 4.0, 5.0},
+                new double[] {0, 3.81966011e-01, 3.0, 5.61803399e+00, 6.0}}
+        };
+    }
+
+    @DataProvider(name = "testDataWithZeroPadding")
+    public Object[][] getTestDataWithZeroPadding() {
+        return new Object[][] {
+                {4,
+                        new double[] {1.0, 1.0, 1.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0}},
+                {4,
+                        new double[] {0.0, 1.0, 1.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0},
+                        new double[] {-1.5, -0.5, 0.5, 1.5}},
+                {4,
+                        new double[] {-1.0, 3.0, 4.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0},
+                        new double[] {-7.5, -3.5, -1.5, 2.5}},
+                {5,
+                        new double[] {1.0, 1.0, 1.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0, 5.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0, 5.0}},
+                {5,
+                        new double[] {1.0, 1.0, 1.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0, 5.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0, 5.0}},
+                {5,
+                        new double[] {0.0, 1.0, 1.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0, 5.0},
+                        new double[] {1.0 - 15.0/8.0, 2.0 - 15.0/8.0, 3.0 - 15.0/8.0, 4.0 - 15.0/8.0, 5.0 - 15.0/8.0}},
+                {5,
+                        new double[] {1.0, 2.0, 1.0},
+                        new double[] {1.0, 2.0, 3.0, 4.0, 5.0},
+                        new double[] {0.39644661, 1.82322330, 4.06066017, 6.73743687, 7.10355339}}
+        };
+    }
+
+    /**
+     * The main test routine
+     * @param dim dimension of the operator
+     * @param fourierFacts Fourier factors
```

Comment:
Really nitpicking (especially since this is test code), but `fourierFacts` sounds more like an instance of `List<SomethingThatIsTrueAboutFourierTransforms>` to me!  I'd just spend the extra two characters on `fourierFactors` (although some might prefer `alternativeFourierFacts`...)